public/locales/en.json

```diff
@@ -0,0 +1,207 @@
+{
```

Comment:
Finding a good naming scheme here is tricky..  

- It's convenient that since this just JSON we can group keys; I would suggest grouping keys by the place in the UI/app they're used -- subgroups should also be used for stuff like notification header/text. A "common" group for text that is often reused is also good to have.  

- For elements that are similar(notifications, errors), I would personally have them all start the same (`notification-xxx`), just so they're easy to parse in the list. 

- Key names themselves should probably all be formatted similarly.
  - For LRR, I use `snake_case` in the backend code, `kebab-case` for CSS and `camelCase` for frontend/javascript.  
  - Since loc keys are more of a frontend styling thing, I would tend to go for kebab-case just to align with CSS? That looks like what you've mostly done already and it feels fine to me. 

So, something like this maybe? I didn't go through everything but hopefully this can be good enough guidance:  
```
"index": {
    "add-archives": "Add Archives",
    "batch-operations": "Batch Operations",
    "settings": "Settings",
    [...]
    "notification-default-password": {
      "heading": "You're using the default password and that's super baka of you",
      "text": "<a href=\"{{loginUrl}}\">Login</a> with password \"kamimamita\" and <a href=\"{{configUrl}}\">change that shit</a> on the double.<br/>...Or just disable it! <br/>Why not check the configuration options afterwards, while you're at it?"
    },
    "notification-debug-mode": {
      "heading": "<i class=\"fas fa-bug\"></i> You're running in Debug Mode!",
      "text": "Advanced server statistics can be viewed <a href=\"{{url}}\">here.</a>"
    },
    "notification-welcome": {
      "heading": "Welcome to LANraragi {{version}}!",
      "text": "If you want to perform advanced operations on an archive, remember to just right-click its name. Happy reading!"
    },
    "context-menu": {
        "read": "Read",
        "download": "Download",
        "edit": "Edit Metadata",
        "delete": "Delete",
        "add-to-category": "Add to Category"
    },
   "error-tag-suggestions": "Couldn't load tag suggestions",
   "error-loading-categories": "Couldn't load categories",
   "error-loading-changelog": "Error getting changelog for new version",
   "error-finding-categories": "Error finding categories for {{id}}!",
},
"settings": {
   "admin-settings": "Admin Settings",
   "select-category": "Select a category to show the matching settings.",
   "save-settings": "Save Settings",
   "plugin-configuration": "Plugin Configuration",
   "database-backup-restore": "Database Backup/Restore",
   "global": {
       "config_global.html.tt2": "------Placeholder------",
       "site-title": "Site Title",
       "site-title-description": "The site title appears on most pages as...their title.",
       "motd-description": "Slang for Message of the Day. Appears on top of the main Library view.",
       [...]
    },
    "security":{
       "enable-password": "Enable Password",
       "password-description": "If enabled, everything that isn't reading will require a password.",
       "new-password": "New Password",
       "new-password-confirm": "New Password Confirmation",
       [...]
    },
    [...]
},
"common": {
    "destructive-operation": "This is a destructive operation!",
    "title": "Title",
    "tags": "Tags",
    "return-to-library": "Return to Library",
}
```

Reply:
This is of course your call, but I'd still like to bike-shed a touch if that's cool.

Grouping/namespacing translations based on location haven't ended up well for me so far. In my experience, it tends to result in duplicate translations and makes it annoying if you move stuff between pages. Grouping on component can work better (like navigation, menus, forms, etc).

Reply:
My thinking behind the "common" group was to avoid duplicating translations, although I suppose it could become a clusterfuck if you end up having a lot of reused strings.  

Grouping on component feels like it'd make building the initial list much harder to write.. In the end I wouldn't really mind either option I suppose

Reply:
I apologize for the recent delays, but I’ve still been working on modifying the relevant code as much as possible.  

I’ve want to continue using my own implementation for text replacement, as the translation methods provided by the official i18n framework don’t quite fit the way I’ve integrated it into my current setup.

Regarding the translated text, I’ve opted to use the "filename" as the parent key for all new entries.  This approach allows for better localization, even though it might result in some redundant translations.